### PR TITLE
Pagination storybook Errorfix

### LINKED ISSUE #200: Frontend: Pagination storybook fails
## User story
**Who:** **As a** developer
**What:** **I want to**fix pagination storybook
**Why:** **so that it** works 
## Acceptance criteria
Pagination does not work when running storybook as you can see in the picture below:
image

### PR DESCRIPTION
**Pull request by suvarna-ratna**
# Description
This PR fix pagination storybook so that it works
Please provide a short summary explaining what this PR is about.

Fixes #200

# How to test?
npm run storybook
see in pagination 

Please provide a short summary how your changes can be tested?

# Checklist

- [X] I have performed a self-review of my own code
- [X] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [X] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [X] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [X] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by Divyajg — COMMENTED** [review]
In the storybook on my machine, I am not able to move the pagination. on click the page numbers not moving to other page numbers!

File: src/client/components/Pagination/Pagination.component.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import './Pagination.styles.css';
```
Comment (FlorentinaPetica) [inline src/client/components/Pagination/Pagination.component.js]: Here you have a check failure, it needs am empty line after import statement not followed by another import.

File: src/client/components/Pagination/Pagination.stories.js [hunk 2]
```diff
@@ -12,15 +12,16 @@ export default {
 
 const template = (args) => {
   // eslint-disable-next-line
-  const [propProducts, setPropProducts] = useState(args.products);
+  const [propProducts, setPropProducts] = useState([]);
```
Comment (FlorentinaPetica) [inline src/client/components/Pagination/Pagination.stories.js:15]: To fix the storybook like Divya is suggesting you can add here another state for currentPageIndex:
`const [currentPageIndex, setCurrentPageIndex] = useState(0);`

File: src/client/components/Pagination/Pagination.stories.js [hunk 3]
```diff
@@ -29,16 +30,13 @@ const template = (args) => {
         </li>
       ))}
       {/* eslint-disable */}
-      <PaginationComponent
-        onPageChange={(range) => setCurrentRange(range)}
-        {...args}
-      />
+      <Pagination onPageChange={(range) => setCurrentRange(range)} {...args} />
```
Comment (FlorentinaPetica) [inline src/client/components/Pagination/Pagination.stories.js]: We need to give pagination the props the it require like this: 
`<Pagination
        currentPageIndex={currentPageIndex}
        setCurrentPageIndex={setCurrentPageIndex}
        pageCount={args.products.length / args.productsPerPage}
        {...args}
      />`

File: src/client/components/Pagination/Pagination.stories.js [hunk 4]
```diff
@@ -12,15 +12,16 @@ export default {
...
   // eslint-disable-next-line
   const [currentRange, setCurrentRange] = useState(
     args.products.slice(0, args.productsPerPage),
   );
   // eslint-disable-next-line
   React.useEffect(() => {
+    setPropProducts(args.products);
     setCurrentRange(propProducts.slice(0, args.productsPerPage));
-  }, [propProducts, args.productsPerPage]);
+  }, [propProducts, args.productsPerPage, args.products]);
```
Comment (FlorentinaPetica) [inline src/client/components/Pagination/Pagination.stories.js]: Here we need to change the setCurrentRange so it slices the products array according to the page index we are on. My solution doesn't look very good but it does the job. If you can change it in a better way it whould be nice. 
`  React.useEffect(() => {
    setPropProducts(args.products);
    setCurrentRange(propProducts.slice(currentPageIndex * args.productsPerPage, args.productsPerPage + (currentPageIndex * args.productsPerPage)));
  }, [propProducts, args.productsPerPage, args.products, currentPageIndex]);`

Comment (FlorentinaPetica) [reply src/client/components/Pagination/Pagination.stories.js]: For me by doing this it shows like this
image

Comment (suvarna-ratna) [reply src/client/components/Pagination/Pagination.stories.js]: Okay .. Thank you Florentina.. I updated as you suggested

File: src/client/components/Pagination/Pagination.stories.js (hunk 3 above)
Comment (suvarna-ratna) [reply src/client/components/Pagination/Pagination.stories.js]: Changed like you mentioned...

File: src/client/components/Pagination/Pagination.stories.js (hunk 2 above)
Comment (suvarna-ratna) [reply src/client/components/Pagination/Pagination.stories.js:15]: done

File: src/client/components/Pagination/Pagination.component.js (hunk 1 above)
Comment (suvarna-ratna) [reply src/client/components/Pagination/Pagination.component.js]: done

**Review by FlorentinaPetica — APPROVED** [review]

**Review by Impressiv3 — APPROVED** [review]
Works in storybook.
Works on the landing page as well.

**Review by dpfernandes — APPROVED** [review]
👍

**Pull request merged**